### ISSUE TITLE
Latest image is not updated using Nexus repository

### ISSUE THREAD

**Issue by hobbsh** [argoproj-labs/argocd-image-updater#111]
**Describe the bug**

The image updater does not seem to pull the latest image properly when using a Nexus image repository. I have enabled trace logs and have not been able to find any threads that would cause this. Looking at the code, it should dump out a list sorted by date and that list in the logs is definitely wrong. I am wondering if this is because creation metadata is buried in Nexus API.

**To Reproduce**

1. Use Nexus image repository to host images
2. Add image updater annotations with `latest` update strategy. I do get different results using `latest-last` and `latest-first` sorting modes on the repository itself, which seems odd. 
3. Observe latest image not being set

**Expected behavior**
Latest image should be used

**Version**
`v0.8.0+961de70`

**Comment by jannfis** [comment]
I do not have access to any Nexus registry. Is there a public Nexus registry available somewhere I could use for testing?

Also, did you try without `latest-first` or `latest-last` options? Without these settings, the meta data is read from the image's manifest - if available (image updater will error if no manifest is returned and neither `latest-first` or `latest-last` is set). If any of this two options is set, image updater assumes the list of tags returned by the registry is sorted by date. 

Unfortunately, every registry implementation out there "does it a little differently". It's really a pity that the API as defined by Docker is a little sparse in its design.

**Comment by hobbsh** [comment]
There is no public Nexus registry that I know of but I believe you should be able to run it in k8s locally with the [chart](https://github.com/helm/charts/tree/master/stable/sonatype-nexus).

It is unfortunate that the docker API does not support tag sorting natively. Good news is that I am actually seeing a logged error now (if I don't use a sorting option):

```
time="2020-10-23T01:53:31Z" level=error msg="Could not get tags from registry: unexpected manifest schema was received from registry: application/vnd.docker.distribution.manifest.v1+json" alias=backend application=my-application image_name=dev/my-application-backend image_tag=a93e858f798723b32ca0dcd1116d51145a7c6022 registry=my-registry.com
```

**Comment by jannfis** [comment]
I think there's a bug or missing feature in the underlying library we use to access the registry API, in that it doesn't fully export the information from a v2 image manifest (i.e. the creation date of the image). Currently, image updater reverts to a v1 image manifest which holds history information. Many registries don't support this v1 scheme anymore as it seems, so probably a fix for this would need to make it upstream. Gonna have a look at it when time allows, but probably this would also allow to use other prominent registries without the `latest-first` and `latest-last` workarounds.

**Comment by jannfis** [comment]
This should be fixed by #115, to be released with v0.8.0

Thanks for reporting it, it gave me some motivation to implement the required changes :)
